main.tf

```diff
@@ -1,13 +1,54 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+resource "aws_wafv2_web_acl_association" "default" {
+  count = module.this.enabled && length(var.association_resource_arns) > 0 ? length(var.association_resource_arns) : 0
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
-  }
+  resource_arn = var.association_resource_arns[count.index]
+  web_acl_arn  = join("", aws_wafv2_web_acl.default.*.arn)
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "aws_wafv2_web_acl_logging_configuration" "default" {
+  count = module.this.enabled && length(var.log_destination_configs) > 0 ? 1 : 0
```

Comment:
If possible, use `for_each` with maps instead of `count` with lists.
When using count with a list, the list changes (e.g. items added or deleted), Terraform will destroy and recreate all the resources.
When using `for_each` with a map, if anything changes in the map, Terraform will update only the changed resources.

(this could be not very critical for WAF rules since we can recreate them anytime, comparing to e.g. RDS instances).

To convert the list to a map, we can do something like 

```
  log_destination_configs_map = {
    for i in var. log_destination_configs : base64encode(i) => i
  }
```

and then use

```
for_each = local.log_destination_configs_map
```

Reply:
I think safely to keep it as that's cos terraform has weird limitation which blocks using of dynamic values inside the `for_each` loop

```sh
The "for_each" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the for_each depends on.
```